### PR TITLE
Update nav_bar.dart

### PR DESCRIPTION
**Pull request by sohrabonline**
On ios devices which ios version is later than , when pop from page to previus page largeTitle showing us bad behavior about its size.

And now, below issues are fixed by this PR:
flutter#67269

flutter#67269

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by MitchellGoodwin** [comment]
Hi @sohrabonline. Thank you for the contribution. As the bot says above, all changes need to have tests attached, in order to defend against regressions. Would you be able to add one? The pre-submit checks seem to be failing as well for some style checks. You should be able to see the failures under the details for the `Linux analyze` test.

**Comment by sohrabonline** [comment]
> Hi @sohrabonline. Thank you for the contribution. As the bot says above, all changes need to have tests attached, in order to defend against regressions. Would you be able to add one? The pre-submit checks seem to be failing as well for some style checks. You should be able to see the failures under the details for the `Linux analyze` test.

can you help me do it?

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 1]
```diff
@@ -878,6 +878,26 @@ class _LargeTitleNavigationBarSliverDelegate
     if (!transitionBetweenRoutes || !_isTransitionable(context)) {
       return navBar;
     }
+
```
Comment (mub-pro) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:881]: 881, 882, 886, and 895 are extra lines you should delete as described in [Linux analyze] failure.

**Review by sohrabonline — COMMENTED** [review]
Extra lines deleted as described in [Linux analyze] failure.

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 2]
```diff
@@ -877,6 +877,22 @@ class _LargeTitleNavigationBarSliverDelegate
       return navBar;
     }
 
+    int _iosSpecificLargeTitleResizer = 0;
```
Comment (mub-pro) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:880]:
```suggestion
    int iosSpecificLargeTitleResizer = 0;
```

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 3]
```diff
@@ -877,6 +877,22 @@ class _LargeTitleNavigationBarSliverDelegate
...
+
+    // this is added for declare ios version as int, i can get it directly, if you can please replace it
+    if (Platform.isIOS) {
+      final _iosVersion = int.tryParse(
+          Platform.operatingSystemVersion.split(' ')[1].split('.').first);
+      _iosSpecificLargeTitleResizer = ((_iosVersion ?? 15) < 14 ? 0 : 4);
+    }
+
+    //and here i have changed fontSize of textStyle for ios 14+ 
+    TextStyle _transitionableNavigationBarTextStyle =
```
Comment (mub-pro) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:890]:
```suggestion
    TextStyle transitionableNavigationBarTextStyle =
```

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 4]
```diff
@@ -877,6 +877,22 @@ class _LargeTitleNavigationBarSliverDelegate
       return navBar;
     }
 
+    int _iosSpecificLargeTitleResizer = 0;
+
+    // this is added for declare ios version as int, i can get it directly, if you can please replace it
+    if (Platform.isIOS) {
+      final _iosVersion = int.tryParse(
```
Comment (mub-pro) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:884]:
```suggestion
      final iosVersion = int.tryParse(
```

Comment (mub-pro) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:884]:
```suggestion
      final int _iosVersion = int.tryParse(
```
   info • Missing type annotation • packages/flutter/lib/src/cupertino/nav_bar.dart:884:7 • always_specify_types

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 5]
```diff
@@ -1,9 +1,9 @@
 // Copyright 2014 The Flutter Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
-
```
Comment (mub-pro) [inline packages/flutter/lib/src/cupertino/nav_bar.dart:4]: The following file does not have the right license header for dart files:

The expected license header is:
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
...followed by a blank line.

**Review by sohrabonline — COMMENTED** [review]
added blank line bottom of header

File: packages/flutter/lib/src/cupertino/nav_bar.dart (hunk 4 above)
Comment (sohrabonline) [reply packages/flutter/lib/src/cupertino/nav_bar.dart]: okay

**Comment by mub-pro** [comment]
@sohrabonline you updated the underscore variables but you still calling them in their previous declaration.

**Review by sohrabonline — COMMENTED** [review]
some variables changed

**Review by sohrabonline — COMMENTED** [review]
name of some private variables changed to public

**Comment by mub-pro** [comment]
@sohrabonline actually, if you go to the checks section in this pr and clicked on (details) link, you see logs for that failure.

**Comment by Hixie** [comment]
@sohrabonline Thanks for your contribution; if you would still like help with writing tests, can you check out the #hackers-new channel on our Discord (link in the contributing guidelines, see below)? Someone there might be able to help. Thanks!

**Comment by MitchellGoodwin** [comment]
@sohrabonline Thank you for putting this together. I'm going to close this PR in order to clear up our review queue. However if you ever want to come back to this, please feel free to reopen at any time.

**Pull request closed without merging**